### DISCUSSION TITLE
Dubious Ownership Git Problem when run the project using Docker

### DISCUSSION THREAD

**Discussion by heliomar-pena** [TeamNovaSoft/qa-exposed-server#8] [Q&A]
When I run the project using Docker I see an error that says:

```
Error: Command failed: git -C /home/user/www/client rev-parse --abbrev-ref HEAD
fatal: detected dubious ownership in repository at '/home/user/www/client'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/user/www/client

    at checkExecSyncError (node:child_process:890:11)
    at execSync (node:child_process:962:15)
    at executeGitCommand (/app/server/utils/git.js:25:10)
    at getCurrentBranch (/app/server/utils/git.js:31:12)
    at /app/server/controllers/home.js:6:31
    at Array.map (<anonymous>)
    at renderHomeView (/app/server/controllers/home.js:5:51)
    at Layer.handle [as handle_request] (/app/node_modules/express/lib/router/layer.js:95:5)
    at next (/app/node_modules/express/lib/router/route.js:144:13)
```

I looked with @Israel-Laguan and found that if you run the command `git config --global --add safe.directory /home/user/www/admin` (where /home/user/www/admin is the absolute path of the folder that have the error) on the scripts of the repository then the problem is fixed.

Also tried to fix it running this command on SH script and Dockerfile but it didn't worked well, maybe it is something about "what user is running the command", or "in what computer", since the problem seems like is happening on docker virtual machine and not host machine.

We need to check what is the best way to fix this error without breaking the U.X and that the user can choose whether to run the commands, or confirm that it does not affect the user's computer.

**Comment by Israel-Laguan** [comment]
Yeah, currently at least on my side I can't test in other system, maybe would need time to check this ownership issue on different environments

**Reply by heliomar-pena** [reply]
I am currently setting up an hyper-v Fedora 39 environment, maybe I could test there

**Reply by heliomar-pena** [reply]
Tested on new Fedora 39 Enviroment
![imagen](https://github.com/TeamNovaSoft/qa-exposed-server/assets/66505715/31073449-6bd7-461a-b8d1-64e63ee417fe)

![imagen](https://github.com/TeamNovaSoft/qa-exposed-server/assets/66505715/0cdc7628-b3cb-49fe-babf-c4b0dbba1edc)

Am receiving error Not such file or directory, but if I execute the same command directly on terminal the directory path is ok

**Reply by heliomar-pena** [reply]
![imagen](https://github.com/TeamNovaSoft/qa-exposed-server/assets/66505715/546ef095-9905-452f-a1f5-606c8cbe10b2)
Also if running program directly instead of use Docker it works fine
